### ISSUE TITLE
esa_enabled is not being treated as optional. 

### ISSUE THREAD

**Issue by pradeep17j** [vmware/terraform-provider-vcf#217]
### Code of Conduct

- [X] I have read and agree to the [Code of Conduct](https://github.com/vmware/terraform-provider-vcf/blob/main/CODE_OF_CONDUCT.md).
- [X] Vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue initial description to help the maintainers prioritize.
- [X] Do not leave "+1" or other comments that do not add relevant information or questions.
- [X] If you are interested in working on this issue or have submitted a pull request, please leave a comment.

### Terraform

v1.7.2

### Terraform Provider

v0.10.0

### VMware Cloud Foundation

5.2

### Description

When `esa_enabled` property is not specified, the validation fails during apply.

```console
### [root@template vcf_bringup]# terraform apply

**
╷
│ Error: JSON Spec Validation
│
│   with vcf_instance.sddc_instance,
│   on main.tf line 4, in resource "vcf_instance" "sddc_instance":
│    4: resource "vcf_instance" "sddc_instance" {
│
│ Invalid vSAN ESA config provided: vSAN ESA config flag cannot be set to empty
│
```

**

### Affected Resources or Data Sources

`resource/vcf_instance`

### vsan section without esa_enabled  
```

      + vsan {
          + datastore_name = "sddc_mgmt_vsan"
          + license        = (sensitive value)
        }
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

vcf_instance.sddc_instance: Creating...
vcf_instance.sddc_instance: Still creating... [10s elapsed]
vcf_instance.sddc_instance: Still creating... [20s elapsed]
╷
│ Error: JSON Spec Validation
│
│   with vcf_instance.sddc_instance,
│   on main.tf line 4, in resource "vcf_instance" "sddc_instance":
│    4: resource "vcf_instance" "sddc_instance" {
│
│ Invalid vSAN ESA config provided: vSAN ESA config flag cannot be set to empty
│

```
 
 ### vsan section with esa_enabled="false" fails at apply

```console
      + vsan {
          + datastore_name = "sddc_mgmt_vsan"
          + esa_enabled    = false
          + license        = (sensitive value)
        }
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

vcf_instance.sddc_instance: Creating...
vcf_instance.sddc_instance: Still creating... [10s elapsed]
vcf_instance.sddc_instance: Still creating... [20s elapsed]
╷
│ Error: JSON Spec Validation
│
│   with vcf_instance.sddc_instance,
│   on main.tf line 4, in resource "vcf_instance" "sddc_instance":
│    4: resource "vcf_instance" "sddc_instance" {
│
│ Invalid vSAN ESA config provided: vSAN ESA config flag cannot be set to empty
│
╵
```

### Debug Output

```console
      + vsan {
          + datastore_name = "sddc_mgmt_vsan"
          + license        = (sensitive value)
        }
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

vcf_instance.sddc_instance: Creating...
vcf_instance.sddc_instance: Still creating... [10s elapsed]
vcf_instance.sddc_instance: Still creating... [20s elapsed]
╷
│ Error: JSON Spec Validation
│
│   with vcf_instance.sddc_instance,
│   on main.tf line 4, in resource "vcf_instance" "sddc_instance":
│    4: resource "vcf_instance" "sddc_instance" {
│
│ Invalid vSAN ESA config provided: vSAN ESA config flag cannot be set to empty
│

```

### Expected Behavior

Successful Bringup

### Actual Behavior

Fails at validation

### Steps to Reproduce

terraform apply

**Comment by tenthirtyam** [comment]
@spacegospod - this was resolved in #203, right?

**Comment by spacegospod** [comment]
@tenthirtyam correct, this is fixed and is set to go out with the next release
